distribution/pull_v2.go

```diff
@@ -403,10 +403,19 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		if err = p.config.ReferenceStore.AddDigest(canonical, imageID, true); err != nil {
 			return false, err
 		}
-	} else if err = p.config.ReferenceStore.AddTag(ref, imageID, true); err != nil {
-		return false, err
-	}
+	} else {
+		if err = p.config.ReferenceStore.AddTag(ref, imageID, true); err != nil {
+			return false, err
+		}
 
+		// Best effort store the digest, this could fail if the digest was
+		// already saved from a different tag.
```

Comment:
This is mostly just an assumption based on reading the code. Haven't had a chance to test it with the error. Can anyone confirm?